src/alternative.rkt

```diff
@@ -3,26 +3,33 @@
 (require "cost.rkt" "programs.rkt")
 (provide (struct-out alt) make-alt alt? alt-expr
          alt-add-event *start-prog* *all-alts*
-         alt-cost alt-equal? alt-map)
+         alt-cost alt-equal? alt-map alt-add-preprocessing
+         make-alt-preprocessing)
 
 ;; Alts are a lightweight audit trail.
 ;; An alt records a low-level view of how Herbie got
 ;; from one program to another.
 ;; They are a labeled linked list of changes.
 
-(struct alt (expr event prevs)
+(struct alt (expr event prevs preprocessing)
         #:methods gen:custom-write
         [(define (write-proc alt port mode)
            (fprintf port "#<alt ~a>" (alt-expr alt)))])
 
 (define (make-alt expr)
-  (alt expr 'start '()))
+  (alt expr 'start '() '()))
+
+(define (make-alt-preprocessing expr preprocessing)
+  (alt expr 'start '() preprocessing))
 
 (define (alt-equal? x y)
   (equal? (alt-expr x) (alt-expr y)))
 
 (define (alt-add-event altn event)
-  (alt (alt-expr altn) event (list altn)))
+  (alt (alt-expr altn) event (list altn) (alt-preprocessing altn)))  
+
+(define (alt-add-preprocessing altn preprocessing)
+  (alt (alt-expr altn) (alt-event altn) (list altn) preprocessing))
```

Comment:
I don't think it will ever bite us in the butt but I would change it like this.

```suggestion
(define (alt-add-preprocessing altn preprocessing)
  (alt (alt-expr altn) 'add-preprocessing (list altn) preprocessing))
```

This might require adding a case in `history.rkt`? Not sure, can advise if you find out.